testsuite/python/gpu_availability.py

```diff
@@ -17,21 +17,61 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 import unittest as ut
+import unittest_decorators as utx
 import espressomd
 
 
 class GPUAvailability(ut.TestCase):
 
     """Tests consistency of GPU availability reporting."""
+    system = espressomd.System(box_l=[1, 1, 1])
 
     def test(self):
         if espressomd.has_features("CUDA"):
-            system = espressomd.System(box_l=[1, 1, 1])
-            self.assertEqual(system.cuda_init_handle.device_list != {},
+            self.assertEqual(self.system.cuda_init_handle.list_devices() != {},
                              espressomd.gpu_available())
+            self.assertEqual(
+                self.system.cuda_init_handle.list_devices_properties() != {},
+                espressomd.gpu_available())
         else:
             self.assertFalse(espressomd.gpu_available())
 
+    @utx.skipIfMissingFeatures("CUDA")
+    def test_exceptions(self):
+        error_msg = 'CUDA error: '
+        if espressomd.gpu_available():
+            n_gpus = len(self.system.cuda_init_handle.list_devices())
+            with self.assertRaisesRegex(RuntimeError, error_msg):
+                self.system.cuda_init_handle.device = n_gpus + 1
+        else:
+            with self.assertRaisesRegex(RuntimeError, error_msg):
+                self.system.cuda_init_handle.device
+            with self.assertRaisesRegex(RuntimeError, error_msg):
+                self.system.cuda_init_handle.device = 0
+
+    @utx.skipIfMissingGPU()
+    def test_devices(self):
```

Comment:
what's the purpose of this test?

Reply:
Not a good test name. It tests the `list_devices*()` methods.